// @progress/kendo-editor-common/dist/npm/main.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var source_1 = require("./source");
exports.getHtml = source_1.getHtml;
exports.setHtml = source_1.setHtml;
exports.parseContent = source_1.parseContent;
exports.trimWhitespace = source_1.trimWhitespace;
var inline_style_1 = require("./inline-style");
exports.applyInlineStyle = inline_style_1.applyInlineStyle;
exports.getInlineStyles = inline_style_1.getInlineStyles;
exports.toggleInlineFormat = inline_style_1.toggleInlineFormat;
var link_1 = require("./link");
exports.applyLink = link_1.applyLink;
exports.removeLink = link_1.removeLink;
var text_1 = require("./text");
exports.insertText = text_1.insertText;
var image_1 = require("./image");
exports.insertImage = image_1.insertImage;
var align_1 = require("./align");
exports.alignBlocks = align_1.alignBlocks;
exports.isAligned = align_1.isAligned;
var cleanFormatting_1 = require("./cleanFormatting");
exports.cleanFormatting = cleanFormatting_1.cleanFormatting;
var blockNode_1 = require("./blockNode");
exports.hasNode = blockNode_1.hasNode;
exports.activeNode = blockNode_1.activeNode;
exports.formatBlockElements = blockNode_1.formatBlockElements;
exports.getBlockFormats = blockNode_1.getBlockFormats;
exports.parentBlockFormat = blockNode_1.parentBlockFormat;
exports.changeTextBlock = blockNode_1.changeTextBlock;
exports.blockNodes = blockNode_1.blockNodes;
exports.cleanTextBlockFormatting = blockNode_1.cleanTextBlockFormatting;
var mark_1 = require("./mark");
exports.hasMark = mark_1.hasMark;
exports.getMark = mark_1.getMark;
exports.getActiveMarks = mark_1.getActiveMarks;
exports.removeAllMarks = mark_1.removeAllMarks;
exports.cleanMarks = mark_1.cleanMarks;
exports.selectionMarks = mark_1.selectionMarks;
var indent_1 = require("./indent");
exports.indent = indent_1.indent;
exports.canIndentAsListItem = indent_1.canIndentAsListItem;
exports.outdent = indent_1.outdent;
exports.canOutdentAsListItem = indent_1.canOutdentAsListItem;
exports.isIndented = indent_1.isIndented;
exports.canBeIndented = indent_1.canBeIndented;
exports.indentBlocks = indent_1.indentBlocks;
var lists_1 = require("./lists");
exports.toggleOrderedList = lists_1.toggleOrderedList;
exports.toggleUnorderedList = lists_1.toggleUnorderedList;
exports.toggleList = lists_1.toggleList;
var utils_1 = require("./utils");
exports.hasSameMarkup = utils_1.hasSameMarkup;
exports.getSelectionText = utils_1.getSelectionText;
exports.getNodeFromSelection = utils_1.getNodeFromSelection;
exports.selectedLineTextOnly = utils_1.selectedLineTextOnly;
exports.canInsert = utils_1.canInsert;
exports.insertNode = utils_1.insertNode;
exports.indentHtml = utils_1.indentHtml;
var align_rules_1 = require("./config/align-rules");
exports.alignLeftRules = align_rules_1.alignLeftRules;
exports.alignCenterRules = align_rules_1.alignCenterRules;
exports.alignRightRules = align_rules_1.alignRightRules;
exports.alignJustifyRules = align_rules_1.alignJustifyRules;
exports.alignRemoveRules = align_rules_1.alignRemoveRules;
var indent_rules_1 = require("./config/indent-rules");
exports.indentRules = indent_rules_1.indentRules;
exports.outdentRules = indent_rules_1.outdentRules;
var schema_1 = require("./config/schema");
exports.nodes = schema_1.nodes;
exports.marks = schema_1.marks;
var keymap_1 = require("./config/keymap");
exports.buildKeymap = keymap_1.buildKeymap;
exports.buildListKeymap = keymap_1.buildListKeymap;
var commands_1 = require("./config/commands");
exports.bold = commands_1.bold;
exports.italic = commands_1.italic;
exports.underline = commands_1.underline;
exports.strikethrough = commands_1.strikethrough;
exports.subscript = commands_1.subscript;
exports.superscript = commands_1.superscript;
exports.link = commands_1.link;
var paste_1 = require("./paste");
exports.sanitize = paste_1.sanitize;
exports.removeComments = paste_1.removeComments;
exports.removeTag = paste_1.removeTag;
exports.pasteCleanup = paste_1.pasteCleanup;
exports.sanitizeClassAttr = paste_1.sanitizeClassAttr;
exports.sanitizeStyleAttr = paste_1.sanitizeStyleAttr;
exports.removeAttribute = paste_1.removeAttribute;
exports.replaceImageSourcesFromRtf = paste_1.replaceImageSourcesFromRtf;
var listConvert_1 = require("./listConvert");
exports.convertMsLists = listConvert_1.convertMsLists;
var table_1 = require("./table");
exports.createTable = table_1.createTable;
var find_replace_1 = require("./find-replace");
exports.find = find_replace_1.find;
exports.findAt = find_replace_1.findAt;
exports.findAll = find_replace_1.findAll;
exports.replace = find_replace_1.replace;
exports.replaceAll = find_replace_1.replaceAll;
var placeholder_1 = require("./plugins/placeholder");
exports.placeholder = placeholder_1.placeholder;
var spaces_fix_1 = require("./plugins/spaces-fix");
exports.spacesFix = spaces_fix_1.spacesFix;
var highlight_1 = require("./plugins/highlight");
exports.textHighlight = highlight_1.textHighlight;
exports.textHighlightKey = highlight_1.textHighlightKey;
// ProseMirror re-exports
tslib_1.__exportStar(require("prosemirror-commands"), exports);
tslib_1.__exportStar(require("prosemirror-dropcursor"), exports);
tslib_1.__exportStar(require("prosemirror-gapcursor"), exports);
tslib_1.__exportStar(require("prosemirror-history"), exports);
tslib_1.__exportStar(require("prosemirror-inputrules"), exports);
tslib_1.__exportStar(require("prosemirror-keymap"), exports);
tslib_1.__exportStar(require("prosemirror-model"), exports);
tslib_1.__exportStar(require("prosemirror-schema-list"), exports);
tslib_1.__exportStar(require("prosemirror-state"), exports);
tslib_1.__exportStar(require("prosemirror-tables"), exports);
tslib_1.__exportStar(require("prosemirror-transform"), exports);
tslib_1.__exportStar(require("prosemirror-view"), exports);
